astropy/units/physical.py

```diff
@@ -630,14 +628,11 @@ def __dir__() -> list[str]:
     ]
 
     for name in sorted(_name_physical_mapping.keys()):
-        physical_type = _name_physical_mapping[name]
+        ptype = _name_physical_mapping[name]
         doclines += [
             f"    * - _`{name}`",
-            f"      - :math:`{physical_type._unit.to_string('latex')[1:-1]}`",
-            f"      - {', '.join([n for n in physical_type if n != name])}",
+            f"      - :math:`{ptype._unit.to_string('latex')[1:-1]}`",
+            f"      - {', '.join([n for n in ptype if n != name])}",
         ]
 
     __doc__ += "\n\n" + "\n".join(doclines)
-
-
-del unit, physical_type
```

Comment:
If we want to keep the module clean, we should now delete `ptype` -- or make that a private variable too.

Reply:
del is probably better, otherwise it hangs around in memory.